### PR TITLE
Implement Integration Registry

### PR DESCRIPTION
**Pull request by TurpIF**
- [x] Declared integration is persisted across sessions
- [x] Different entry points declare themselves to the registry
- [x] Provide a method to retrieve profile ID
- [x] Mediation adapters are detected
- [x] If mediation are detected, mediation is kept even if standalone declare itself
- [x] If no info is available (for prefetch, remote config) use fallback profile ID = 235

JIRA: EE-1195

### CODE REVIEW

File: publisher-sdk/src/main/java/com/criteo/publisher/integration/IntegrationRegistry.kt [hunk 1]
```diff
@@ -17,8 +17,8 @@
 package com.criteo.publisher.integration
 
 import android.content.SharedPreferences
-import com.criteo.publisher.annotation.OpenForTesting
 import androidx.annotation.VisibleForTesting
+import com.criteo.publisher.annotation.OpenForTesting
```
Comment (vguerci) [inline publisher-sdk/src/main/java/com/criteo/publisher/integration/IntegrationRegistry.kt:21]: Looks unrelated to its commit

File: publisher-sdk/src/main/java/com/criteo/publisher/model/Config.java [hunk 2]
```diff
@@ -70,7 +68,7 @@
   private volatile RemoteConfigResponse cachedRemoteConfig;
 
   @Nullable
-  private final Context context;
+  private final SharedPreferences sharedPreferences;
```
[the remark this answers is not in the document]
Comment (TurpIF) [reply publisher-sdk/src/main/java/com/criteo/publisher/model/Config.java:71]: I extracted this change in a dedicated to make it nicer to review

File: build.gradle.kts [hunk 3]
```diff
@@ -20,6 +20,7 @@ buildscript {
   dependencies {
     classpath(Deps.Android.GradlePlugin)
     classpath(Deps.Kotlin.GradlePlugin)
+    classpath(Deps.Kotlin.AllOpenPlugin)
```
[the remark this answers is not in the document]
Comment (TurpIF) [reply build.gradle.kts:23]: I extracted this change in a dedicated commit, and added documentation on the configured annotation.

File: publisher-sdk/src/main/java/com/criteo/publisher/integration/IntegrationRegistry.kt (hunk 1 above)
Comment (TurpIF) [reply publisher-sdk/src/main/java/com/criteo/publisher/integration/IntegrationRegistry.kt:21]: Maybe now it seems more related => as test need the class to be opened

**Comment by vguerci** [comment]
Thanks for splitting / commenting the commits, please next time try to do this before review ;)
